CONTRIBUTING.rst

```diff
@@ -53,6 +47,9 @@ General coding advices:
 -  Block comments SHOULD reference the code following them and SHOULD be
    indented to the same level
 
+Also checkout ``REVIEW_CHECKLIST`` to see what code reviewers will look for
```

Comment:
I personally think it would be simpler to merge the content of `REVIEW_CHECKLIST.rst` here, since it is weird to check the coding guidelines in two different places. I Ilke `CONTRIBUTING` as the file name because is more general than `CODING_GUIDELINES`  but it can be changed to something else if you think is misleading. Also, in my opinion most the items in the review guidelines are actually coding guidelines, and thus the review guidelines could basically say, check the coding guidelines and additionally check the legacy and the legal stuff about the contribution agreeement, or something like that.

Reply:
I personally do not care about naming. However I prefer having a document for people who are looking into the posibility of contributing, which 1) welcomes potential contributors, 2) explains the process for contributing (pick / write an issue -> make a PR -> get a review), 3) explains that you have to sign an agreement, 4) links to but does not scare people off with in-detail coding / legal / review guidelines.

Reply:
If we can agree on that, I would propose to instead move all coding guidelines out of the `CONTRIBUTING`. Because on one hand, as you say that name is more general and probably the first one I would look at. And because as you also pointed out it is better to have all the coding guidelines in one place.

Reply:
Sure, that also makes sense. As long as all the coding-related information is together in a single file instead of split in different places, it's fine for me